server/pxf-json/src/test/java/org/greenplum/pxf/plugins/json/parser/JsonRecordReaderTest.java

```diff
@@ -0,0 +1,355 @@
+        jsonRecordReader = new JsonRecordReader(jobConf, fileSplit);
+
+        assertEquals(0, jsonRecordReader.getPos());
+
+        key = createKey();
+        data = createValue();
+        jsonRecordReader.next(key, data);
+        // reads the full 1 record here
+        assertEquals(105, data.toString().getBytes(StandardCharsets.UTF_8).length);
+
+        assertEquals("{\"cüstömerstätüs\":\"välid\",\"name\": \"äää\", \"year\": \"2022\", \"address\": \"söme city\", \"zip\": \"95051\"}", data.toString());
+    }
+
+    @Test
+    /**
+     * The Split size is large only single spilt will be able to read all the records.
+     */
+    public void testRecordSizeSmallerThanSplit() throws IOException {
+
+        fileSplit = new FileSplit(path, 0, 1000, hosts);
+        jsonRecordReader = new JsonRecordReader(jobConf, fileSplit);
+
+        assertEquals(0, jsonRecordReader.getPos());
+
+        key = createKey();
+        data = createValue();
+        int recordCount = 0;
+        while (jsonRecordReader.next(key, data)) {
+            assertNotNull(data);
+            recordCount++;
+        }
+
+        assertEquals(5, recordCount);
+
+        // assert the last record json
+        assertEquals("{\"cüstömerstätüs\":\"välid\",\"name\": \"你好\", \"year\": \"2033\", \"address\": \"0\uD804\uDC13a\", \"zip\": \"19348\"}", data.toString());
+    }
+
+    @Test
+    public void testEmptyFile() throws URISyntaxException, IOException {
+
+        file = new File(this.getClass().getClassLoader().getResource("parser-tests/offset/empty_input.json").toURI());
+        path = new Path(file.getPath());
+        fileSplit = new FileSplit(path, 0, 10, hosts);
+        jsonRecordReader = new JsonRecordReader(jobConf, fileSplit);
+
+        assertEquals(0, jsonRecordReader.getPos());
+
+        key = createKey();
+        data = createValue();
+        int recordCount = 0;
+        if (jsonRecordReader.next(key, data)) {
+            assertNotNull(data);
+            recordCount++;
+        }
+
+        assertEquals(0, recordCount);
+        assertEquals(0, jsonRecordReader.getPos());
+        assertEquals("", data.toString());
+    }
+
+    @Test
+    public void testEmptyJsonObject() throws URISyntaxException, IOException {
+
+        file = new File(this.getClass().getClassLoader().getResource("parser-tests/offset/empty_json_object.json").toURI());
+        path = new Path(file.getPath());
+        fileSplit = new FileSplit(path, 0, 10, hosts);
+        jsonRecordReader = new JsonRecordReader(jobConf, fileSplit);
+
+        assertEquals(0, jsonRecordReader.getPos());
+
+        key = createKey();
+        data = createValue();
+        int recordCount = 0;
+        if (jsonRecordReader.next(key, data)) {
+            assertNotNull(data);
+            recordCount++;
+        }
+
+        assertEquals(0, recordCount);
+        assertEquals(12, jsonRecordReader.getPos());
+        assertEquals("", data.toString());
```

Comment:
```suggestion
        assertFalse(jsonRecordReader.next(key, value));
        assertEquals(12, jsonRecordReader.getPos());
```

Since `jsonRecordReader.next(key, value)` is only supposed to return `true` if a key/value was read, I don't know if we need to assert anything about `data`.